### PR TITLE
indexTable sortable

### PR DESCRIPTION
**Pull request by vvalentin-lucca**
## Description

New design makes it possible to differentiate between sortable and non-sortable columns without having to fly over them.

-----

Sortable table code is common to both `.table` and `.indexTable` components.

-----

Capture d’écran 2024-06-14 à 11 24 14

Capture d’écran 2024-06-11 à 17 52 45

### CODE REVIEW

**Comment by LuccaIntegration** [comment]
:woman_cook: https://lucca-front.lucca.tech/PR-2858/storybook

**Comment by CCNET-iLucca** [comment]
[Tests d'interfaces](https://qa.lucca.tech/18244/ui-diff)

File: packages/scss/src/components/indexTableSorted/component.scss [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+    --components-button-padding: 0 var(--pr-t-spacings-50);
+    --components-button-gap: var(--pr-t-spacings-25);
+
+    user-select: none;
+    font-weight: 400;
+    vertical-align: top;
+    align-items: flex-start;
+    left: var(--components-sorted-offset);
+    text-align: inherit;
+    flex-direction: var(--components-sorted-direction);
```
Comment (fbasmaison-lucca) [inline packages/scss/src/components/indexTableSorted/component.scss:17]: Indentation is too large here (5-17).

Comment (fbasmaison-lucca) [reply packages/scss/src/components/indexTableSorted/component.scss:17]: Actually, this file hesitate between 2 spaces and 4 spaces indentation.

(Maybe adding some tabs would be fun! :p)

File: packages/scss/src/components/indexTableSorted/vars.scss [hunk 2]
```diff
@@ -0,0 +1,6 @@
+@mixin vars {
+	--components-sorted-arrowColor: var(--palettes-neutral-500);
+    --components-sorted-arrowScale: 1;
+    --components-sorted-direction: row;
+    --components-sorted-offset: calc(var(--pr-t-spacings-50) * -1);
```
Comment (fbasmaison-lucca) [inline packages/scss/src/components/indexTableSorted/vars.scss:5]: Inconsistent indentation here as well.

**Comment by CCNET-iLucca** [comment]
[Tests d'interfaces](https://qa.lucca.tech/18260/ui-diff)

**Comment by LuccaIntegration** [comment]
:woman_cook: https://lucca-front.lucca.tech/PR-2858/storybook

**Comment by CCNET-iLucca** [comment]
[Tests d'interfaces](https://qa.lucca.tech/18331/ui-diff)

**Comment by LuccaIntegration** [comment]
:woman_cook: https://lucca-front.lucca.tech/PR-2858/storybook

**Comment by CCNET-iLucca** [comment]
[Tests d'interfaces](https://qa.lucca.tech/21282/ui-diff)

File: packages/scss/src/components/tableSorted/component.scss [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+	user-select: none;
+	font-weight: 400;
+	vertical-align: top;
+	align-items: flex-start;
+	left: var(--components-sorted-offset);
+	text-align: inherit;
+	flex-direction: var(--components-sorted-direction);
+
+	@at-root ($atRoot) {
+		.table-head-row-cell-sortableButton-arrows,
```
Comment (jeremie-lucca) [inline packages/scss/src/components/tableSorted/component.scss:20]: If we consider tableSorted as a component, maybe could replace `.table-head-row-cell-sortableButton` and `.indexTable-head-row-cell-sortableButton` basis by `.tableSortable` ?

**Review by jeremie-lucca — CHANGES_REQUESTED** [review]

**Comment by LuccaIntegration** [comment]
:woman_cook: https://lucca-front.lucca.tech/PR-2858/storybook

**Comment by CCNET-iLucca** [comment]
[Tests d'interfaces](https://qa.lucca.tech/21740/ui-diff)

**Review by jeremie-lucca — APPROVED** [review]

**Pull request merged**